frontend/src/metabase/components/tree/TreeNode.jsx

```diff
@@ -27,65 +27,72 @@ const propTypes = {
   variant: PropTypes.string,
 };
 
-export const TreeNode = React.memo(function TreeNode({
-  isExpanded,
-  isSelected,
-  hasChildren,
-  onToggleExpand,
-  onSelect,
-  depth,
-  item,
-  variant,
-}) {
-  const { name, icon, hasRightArrow, id } = item;
+// eslint-disable-next-line react/display-name
+export const TreeNode = React.memo(
+  React.forwardRef(function TreeNode(
+    {
+      isExpanded,
+      isSelected,
+      hasChildren,
+      onToggleExpand,
+      onSelect,
+      depth,
+      item,
+      variant,
+    },
+    ref,
+  ) {
+    const { name, icon, hasRightArrow, id } = item;
```

Comment:
Should we add the display name explicitly? Example:

```js
export const TreeNode = React.memo(React.forwardRef(() => <TreeNode />))

TreeNode.displayName = "TreeNode"
TreeNode.propTypes = propTypes
```

Reply:
@kulyk unfortunately, it shows the warning I had to suppress because of nested `React.forwardRef` in `React.memo`. The upper `TreeNode` does not really need to have `displayName` defined explicitly here:
<img width="347" alt="Screen Shot 2021-07-12 at 14 49 29" src="https://user-images.githubusercontent.com/14301985/125282786-6ea99c00-e320-11eb-8f76-56268bd02d74.png">